src/main/java/be/seeseemelk/mockbukkit/block/data/BlockDataMock.java

```diff
@@ -0,0 +1,61 @@
+package be.seeseemelk.mockbukkit.block.data;
+
+import org.bukkit.Material;
+import org.bukkit.block.data.BlockData;
+
+import org.jetbrains.annotations.NotNull;
+import org.jetbrains.annotations.Nullable;
+
+public class BlockDataMock implements BlockData
+{
+    private final Material type;
+
+    public BlockDataMock(Material type)
+    {
+        this.type = type;
+    }
+
+    @Override
+    public @NotNull Material getMaterial()
+    {
+        return type;
+    }
+
+    @Override
+    public @NotNull String getAsString()
+    {
+        // TODO Auto-generated method stub
+        throw new UnsupportedOperationException();
```

Comment:
Maybe change these from `UnsupportedOperationException` to `UnimplementedOperationException`. The latter type of exception will cause JUnit to not show any test using the function as having a failure, but rather to be ignored as it uses functionality that cannot be tested yet.